src/Microsoft.ML.StaticPipe/StaticSchemaShape.cs

```diff
@@ -280,7 +280,8 @@ private static Type GetTypeOrNull(Schema.Column col)
                     {
                         var kvcol = meta.Schema[kvcolIndex];
                         var kvType = kvcol.Type;
-                        if (kvType is VectorType kvVecType && kvVecType.Size == kt.Count)
+                        Contracts.Assert(kt.Count <= int.MaxValue);
```

Comment:
>Assert [](start = 34, length = 6)

Is it fine to have an Assert here?

Reply:
This assert is fine. `kvVecType.Size` is integer, and above you check whether `kt.Count` equals it. So, it must be in the range of an `int`. So, what you are doing here is you're asserting to make clear to the reader that your `(int)` cast below is fine, which seems like a perfectly fine use of an assert for code documentation @artidoro .